src/test/java/test/JUnitTestClassLoader.java

```diff
@@ -95,14 +97,15 @@ private Listener runJunitTest(String src, String name) throws Exception {
 
   private Class<?> compile(String src, String name) throws Exception {
     // compile class and load it into by a custom classloader
-    File srcFile = new File(tmpDir, name + ".java");
+    File directory = createTempDir().toFile();
+    File srcFile = new File(directory, name + ".java");
```

Comment:
You should have a look at
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createTempFile-java.nio.file.Path-java.lang.String-java.lang.String-java.nio.file.attribute.FileAttribute...-

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newBufferedReader-java.nio.file.Path-

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newInputStream-java.nio.file.Path-java.nio.file.OpenOption...-

Reply:
Cannot use `Files.createTempFile` because it doesn't allow me to specify the file name. But used the other two links to refactor the code and made use of the new APIs in `Files` class.

Reply:
Right, but we don't care about the file name. We just need to know it and maybe to specify a suffix `.java`.

BTW, it is not so important.